runtime/tools/python/ttrt/common/golden.py

```diff
@@ -127,27 +128,40 @@ def golden(binary, programContext, opContext):
         op_golden_tensor = binary.get_debug_info_golden(loc)
         op_output_tensor = ttrt.runtime.get_op_output_tensor(opContext, programContext)
 
-        if len(op_golden_tensor) == 0:
-            print("Golden tensor is empty - skipping golden comparison")
+        if op_golden_tensor is None:
+            print("Golden tensor is None - skipping golden comparison")
             return
 
         if len(op_output_tensor) == 0:
             print("Output tensor is empty - skipping golden comparison")
             return
 
-        if len(op_golden_tensor) != len(op_output_tensor):
+        # Determine the element type of the golden tensor
+        match op_golden_tensor.dtype:
```

Comment:
So technically on macOS we only support python 3.9, but then again this code will only run on runtime enabled builds i.e. linux which has python 3.10.  We'll see what CI says :)

Reply:
CI seems fine with it, if we run into an issue we can switch this to a map or something.